modules/git/git_test.go

```diff
@@ -1,14 +1,85 @@
 package git
 
 import (
+	"fmt"
+	"os"
+	"os/exec"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestGetCurrentBranchName(t *testing.T) {
-	t.Parallel()
+var branchNameMock string
+var tagMock string
```

Comment:
As mentioned above, we will want to test if the `git` command is correct. We are more likely to get the `git` command wrong than the surrounding logic, so it's important that we run the actual command instead of mocking it out.

Reply:
Removed mock, switched to temp repo cloning strategy